### DISCUSSION TITLE
Is this valid? No errors are reported

### DISCUSSION THREAD

**Discussion by markfirmware** [alpinejs/alpine#978] [1. Help] [answered]
https://codepen.io/markfirmware/pen/YzGLPXX

x-if causes an error with no expression - should x-for do the same?

**Comment by KevinBatdorf** [comment, marked as answer]
`x-for` requires an expression. See the docs: https://github.com/alpinejs/alpine#x-for

Here's the errors I see. Not the most useful but you can jump to the code and identify quickly enough that it's related to `x-for`

<img width="1226" alt="Screen Shot 2021-01-04 at 11 58 06 PM" src="https://user-images.githubusercontent.com/1478421/103559447-b91d0c80-4ee8-11eb-9863-528250cdb098.png">

**Reply by markfirmware** [reply]
![Screenshot from 2021-01-04 12-09-01](https://user-images.githubusercontent.com/22677068/103560489-c36ee480-4eaf-11eb-9989-4450a6e60562.png)

**Reply by KevinBatdorf** [reply]
The CodePen Console isn't part of the browser but something CodPen added. It's not full featured so you shouldn't rely on it. It's just for simple logging.

**Reply by markfirmware** [reply]
Swell. I can use c-s-k but then I lose the run button. If I set preview-on-save then I can use c-s but sometimes need an artificial change to effect the preview. Good enough.
